lib/ancestry/materialized_path.rb

```diff
@@ -75,11 +75,10 @@ def siblings_of(object)
     end
 
     def ordered_by_ancestry(order = nil)
-      if %w(mysql mysql2 sqlite sqlite3 postgresql).include?(connection.adapter_name.downcase) && ActiveRecord::VERSION::MAJOR >= 5
-        reorder(
-          Arel::Nodes::Ascending.new(Arel::Nodes::NamedFunction.new('COALESCE', [arel_table[ancestry_column], Arel.sql("''")])),
-          order
-        )
+      if %w(mysql mysql2 sqlite sqlite3).include?(connection.adapter_name.downcase)
+        reorder(arel_table[ancestry_column], order)
+      elsif %w(postgresql).include?(connection.adapter_name.downcase) && ActiveRecord::VERSION::STRING >= "6.1"
+        reorder(Arel::Nodes::Ascending.new(arel_table[ancestry_column]).nulls_first)
       else
         reorder(Arel.sql("(CASE WHEN #{connection.quote_table_name(table_name)}.#{connection.quote_column_name(ancestry_column)} IS NULL THEN 0 ELSE 1 END), #{connection.quote_table_name(table_name)}.#{connection.quote_column_name(ancestry_column)}"), order)
```

Comment:
ok, thinking this one through.

Before this change, the 3 vendors worked.
Now after this change, which was targeting mysql, postgres is broken.

So just adding line 78 and 79 to the top (and 80-81) would fix this.

We have the coalesce and case statements because sybase, oracle, and microsoft sql server were holdouts for the coalesce statement. They had `ISNULL` or `NVL` or other to do the same thing and it took a while for the ANSI stander to get into the various platforms. IBM DB2 was probably one of the first converts.

This is many many years later, so since we don't care about active support 4.2, I'm thinking we can drop our `else` block (83) and use the previous code for postgres (deleted 79-82)

```suggestion
        reorder(
          Arel::Nodes::Ascending.new(Arel::Nodes::NamedFunction.new('COALESCE', [arel_table[ancestry_column], Arel.sql("''")])),
          order
        )
```

The final diff should probably show this addition as unchanged rather than an addition.